File: oaer/lib/api/nats.ts
/*
 * nats.ts
 * Copyright (C) 2023 veypi <i@veypi.com>
 * 2023-10-19 21:36
 * Distributed under terms of the MIT license.
 */


import cfg from '../cfg'
import ajax from './axios'

export default {
  local: () => cfg.BaseUrl() + '/nats/',
  general() {
    return ajax.get(this.local() + 'varz')
  },
  conns() {
    return ajax.get(this.local() + 'connz', { subs: true })
  },
}

File: oa/cfg/nats.go
//
// nats.go
// Copyright (C) 2024 veypi <i@veypi.com>
// 2024-10-11 17:32
// Distributed under terms of the MIT license.
//

package cfg

import (
	"github.com/nats-io/nats-server/v2/server"
	"github.com/veypi/utils/logv"
)

func RunNats() {
	opts := &server.Options{
		ConfigFile: "~/.config/oa/nats.cfg2",
	}
	ns, err := server.NewServer(opts)
	if err != nil {
		panic(err)
	}
	// Start the nats server
	logv.Info().Msg("nats server start")
	ns.Start()
}
